src/local/terrain.rs

```diff
@@ -0,0 +1,58 @@
+use std::mem::MaybeUninit;
+
+use js_sys::Uint8Array;
+
+use crate::{constants::Terrain, objects::RoomTerrain};
+
+use super::{xy_to_terrain_index, RoomXY, ROOM_AREA};
+
+pub struct LocalRoomTerrain {
```

Comment:
This struct should at least be `Debug`, as is suggested for *all* structs.  Additionally, should it be `Clone` and (de)serializable?
```suggestion
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LocalRoomTerrain {
```